live-teaching-web/src/app/register/register.component.ts

```diff
@@ -29,14 +29,13 @@ export class RegisterComponent implements OnInit {
     private router: Router,
     private afAuth: AngularFireAuth,
     private authService: AuthService,
-    private messageService: MessageService
   ) {}
 
   ngOnInit() {
     this.afAuth.user.subscribe(user => {
       if (user) {
-        this.userAuth = user.providerData[0];
-        let id = this.userAuth.email.split("@psu.ac.th")[0];
+        this.userAuth = user;
+        let id = this.userAuth.providerData[0].email.split("@psu.ac.th")[0];
```

Comment:
If providerData is `null`. bug ?

Reply:
already check before route to this page but not use now